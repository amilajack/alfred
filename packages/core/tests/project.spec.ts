import path from 'path';
import Project from '../src/project';

describe('Project', () => {
  describe('dependencies', () => {
    let depsToInstall;

    beforeAll(async () => {
      const defaultProject = new Project(
        path.join(__dirname, 'fixtures/react-app')
      );
      depsToInstall = await defaultProject.findDepsToInstall();
    });

    it('should include skills in config', () => {
      expect(depsToInstall.devDependencies).toHaveProperty('react');
    });

    it('should match snapshot', () => {
      expect(depsToInstall).toMatchSnapshot();
    });

    it('should not include deps already in package.json', () => {
      expect(depsToInstall.devDependencies).not.toHaveProperty('react-dom');
    });
  });
});
