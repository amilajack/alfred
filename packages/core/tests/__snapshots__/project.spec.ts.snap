// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project dependencies should match snapshot 1`] = `
Object {
  "dependencies": Object {},
  "devDependencies": Object {
    "@babel/cli": "^7.8.3",
    "@babel/core": "^7.8.3",
    "@babel/plugin-proposal-class-properties": "^7.8.3",
    "@babel/plugin-transform-react-constant-elements": "^7.0.0",
    "@babel/plugin-transform-react-inline-elements": "^7.0.0",
    "@babel/preset-env": "^7.8.3",
    "@babel/preset-react": "7.0.0",
    "@babel/register": "^7.8.3",
    "babel-eslint": "10.0.0",
    "babel-loader": "8.0.6",
    "babel-plugin-dev-expression": "^0.2.1",
    "babel-plugin-transform-react-remove-prop-types": "^0.4.20",
    "eslint": "^6.8.0",
    "eslint-config-airbnb": "18.0.0",
    "eslint-config-prettier": "6.9.0",
    "eslint-plugin-jest": "23.6.0",
    "eslint-plugin-prettier": "3.0.1",
    "jest": "^24.0.0",
    "parcel": "^1.11.0",
    "prettier": "^1.16.4",
    "react": "16.0.0",
    "react-hot-loader": "^4.3.12",
  },
}
`;
